pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -72,38 +136,55 @@ in buildPythonPackage rec {
   PYTORCH_BUILD_VERSION = version;
   PYTORCH_BUILD_NUMBER = 0;
 
+  BUILD_NAMEDTENSOR = buildNamedTensor;  # experimental feature
+  USE_SYSTEM_NCCL=true;                  # don't build pytorch's third_party NCCL
+
   # Suppress a weird warning in mkl-dnn, part of ideep in pytorch
   # (upstream seems to have fixed this in the wrong place?)
   # https://github.com/intel/mkl-dnn/commit/8134d346cdb7fe1695a2aa55771071d455fae0bc
-  NIX_CFLAGS_COMPILE = lib.optionals (numpy.blasImplementation == "mkl") [ "-Wno-error=array-bounds" ];
+  # https://github.com/pytorch/pytorch/issues/22346
+  #
+  # Also of interest: pytorch ignores CXXFLAGS uses CFLAGS for both C and C++:
+  # https://github.com/pytorch/pytorch/blob/v1.2.0/setup.py#L17
+  NIX_CFLAGS_COMPILE = lib.optionals (my_numpy.blasImplementation == "mkl") [ "-Wno-error=array-bounds" ];
 
   nativeBuildInputs = [
-     cmake
-     utillinux
-     which
+    cmake
+    utillinux
+    which
+    ninja
   ] ++ lib.optionals cudaSupport [ cudatoolkit_joined ];
 
   buildInputs = [
-     numpy.blas
-  ] ++ lib.optionals cudaSupport [ cudnn ]
+    my_numpy.blas
+  ] ++ lib.optionals cudaSupport [ cudnn my_magma nccl ]
     ++ lib.optionals stdenv.isLinux [ numactl ];
 
   propagatedBuildInputs = [
     cffi
-    numpy
+    my_numpy
     pyyaml
-  ] ++ lib.optional (pythonOlder "3.5") typing;
-
-  checkInputs = [ hypothesis ];
-  checkPhase = ''
-    ${cudaStubEnv}python test/run_test.py --exclude dataloader sparse torch utils thd_distributed distributed cpp_extensions
+  ] ++ lib.optionals openMPISupport [ my_openmpi ]
+    ++ lib.optional (pythonOlder "3.5") typing;
+
+  checkInputs = [ hypothesis ninja ];
+  checkPhase = "${cudaStubEnv}python test/run_test.py"
+    + " --exclude utils" # utils requires git, which is not allowed in the check phase
+
+    # Other tests which have been disabled in previous nix derivations of pytorch.
```

Comment:
If these tests pass now I thin you can remove?

Reply:
Sure, but maybe a better way to do this would be to leave these and pass along a `doCheck = false`? Would that work? I'm still trying to figure out the python-nix-ecosystem. I do want to make sure that people can verify that everything is functional on their box if desired.

Reply:
I've added `doCheck=false;` above. Let me know if this isn't enough (maybe I misunderstand doCheck)

Reply:
little OT: I suspect that with `doCheck=false;` the build will succeed on Darwin but then will fail later at runtime ;)

Reply:
@stites I merged this commit into my `nix` branch, and it had the desired (?) effect of skipping all tests for pytorch.